build.yaml

```diff
@@ -4038,22 +4038,36 @@ steps:
     image: mcr.microsoft.com/azure-cli
     alwaysRun: true
     script: |
-      set -e
-      AZURE_USERNAME=$(jq -r '.appId' /test-gsa-key/key.json)
+      set -e -o pipefail
+      AZURE_USERNAME=$(jq -r '.app Id' /test-gsa-key/key.json)
       AZURE_PASSWORD=$(jq -r '.password' /test-gsa-key/key.json)
       AZURE_TENANT_ID=$(jq -r '.tenant' /test-gsa-key/key.json)
-      AZURE_RESOURCE_GROUP="{{ global.azure_resource_group }}"
       az login --service-principal -u $AZURE_USERNAME -p $AZURE_PASSWORD --tenant $AZURE_TENANT_ID
-      set +e
-      az vm list --resource-group $AZURE_RESOURCE_GROUP -o tsv \
-          --query "[?tags.namespace == '{{ default_ns.name }}'].name"
-        | xargs -n1 -r sh -c 'az vm delete --resource-group $AZURE_RESOURCE_GROUP --name "$1" || true' argv0
-      az network nic list --resource-group $AZURE_RESOURCE_GROUP -o tsv \
-          --query "[?tags.namespace == '{{ default_ns.name }}'].name"
-        | xargs -n1 -r sh -c 'az network nic delete --resource-group $AZURE_RESOURCE_GROUP --name "$1" || true' argv0
-      az network public-ip list --resource-group $AZURE_RESOURCE_GROUP -o tsv \
-          --query "[?tags.namespace == '{{ default_ns.name }}'].name"
-        | xargs -n1 -r sh -c 'az network public-ip delete --resource-group $AZURE_RESOURCE_GROUP --name "$1" || true' argv0
+      set -x
+
+      failures=0
+
+      AZURE_RESOURCE_GROUP="{{ global.azure_resource_group }}"
+      vm_names=$(az vm list --resource-group $AZURE_RESOURCE_GROUP -o tsv --query "[?tags.namespace == '{{ default_ns.name }}'].name")
+      for name in $vm_names
+      do
+          set +e
+          az vm delete --resource-group $AZURE_RESOURCE_GROUP --name $name --yes
+          vm_exit_code=$?
+          az network nic delete --resource-group $AZURE_RESOURCE_GROUP --name $name
+          nic_exit_code=$?
+          az network public-ip delete --resource-group $AZURE_RESOURCE_GROUP --name $name
+          ip_exit_code=$?
+          set -e
+
+          if [[ $vm_exit_code -ne 0 || $nic_exit_code -ne 0 || $ip_exit_code -ne 0 ]]
```

Comment:
I'm a bit worried about this code. You can't delete a nic or an ip that are attached to the VM. The subsequent commands should error if the VM existed.